metaflow/plugins/aip/argo_utils.py

```diff
@@ -14,16 +14,38 @@
 
 def run_argo_workflow(
     kubernetes_namespace: str,
-    template_name: str,
+    template_name: Optional[str] = None,
+    project_name: Optional[str] = None,
+    branch_name: Optional[str] = None,
+    template_name_prefix: Optional[str] = None,
     parameters: Optional[dict] = None,
     wait_timeout: Union[int, float, datetime.timedelta] = 0,
     **kwarg,  # Other parameters for wait function
 ) -> Tuple[str, str]:
+    """
+    Using template_name to trigger a workflow template name with exact match.
+
+    If no template_name is provided, the latest workflow template satisfying
+    the project_name, branch_name and template_name_prefix will be used.
+    All of these filters are optional. If not provided, the latest workflow
+    from the namespace will be used.
+    """
+    client = ArgoClient(namespace=kubernetes_namespace)
+
+    if not template_name:
+        template_name = get_latest_workflow(
+            kubernetes_namespace,
+            project_name=project_name,
+            branch_name=branch_name,
+            template_name_prefix=template_name_prefix,
+        )
```

Comment:
it'd be great to test this scenario in https://github.com/zillow/metaflow/blob/0a913d1e0879e85c53b11587e9a26fa06213e93e/metaflow/plugins/aip/tests/flows/flow_triggering_flow.py#L87 (which could be another ticket to unblock this merge)

Reply:
I'll add tests in the same MR. I just want to get comments from you before fully populating the test cases.

Reply:
Tests added.